usecases/memwatch/monitor.go

```diff
@@ -117,9 +121,39 @@ func (m *Monitor) updateLimit() {
 	m.limit = m.limitSetter(-1)
 }
 
+func NewDummyMonitor() *Monitor {
+	m := &Monitor{
+		metricsReader: func() int64 { return 0 },
+		limitSetter:   func(size int64) int64 { return TiB },
+		maxRatio:      1,
+	}
+	m.Refresh()
+	return m
+}
+
 type metricsReader func() int64
 
 type AllocChecker interface {
 	CheckAlloc(sizeInBytes int64) error
 	Refresh()
 }
+
+func EstimateObjectMemory(object *models.Object) int64 {
+	// Note: This is very much oversimplified. It assumes that we always need
+	// the footprint of the full vector and it assumes a fixed overhead of 30B
+	// per vector. In reality this depends on the HNSW settings - and possibly
+	// in the future we might have completely different index types.
+	//
+	// However, in the meantime this should be a fairly reasonable estimate, as
+	// it's not meant to fail exactly on the last available byte, but rather
+	// prevent OOM crashes. Given the fuzziness and async style of the
+	// memtracking somewhat decent estimate should be good enough.
+	return int64(len(object.Vector)*4 + 30)
+}
+
+func EstimateObjectDeleteMemory() int64 {
+	// When deleting an object we attach a tombstone to the object in the HNSW and a new segment in the Memtable and
+	// additional other temporary allocations.
+	// The total amount is hard to guess so we go with 100 bytes.
+	return 100
```

Comment:
Also don't have a solid estimation for this, we can always just adjust as needed. Since it is just a wild guess, an idea is to just make this configurable via env var, and default to `100`. This way we don't need another PR to adjust if this value needs to be changed